x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java

```diff
@@ -426,7 +427,19 @@ protected void doRun() {
                     e -> closeHandler.accept(e, true)
                 ));
             },
-            e -> closeHandler.accept(e, true));
+            e -> {
+                if (ExceptionsHelper.unwrapCause(e) instanceof InvalidAliasNameException) {
+                    String msg = "Detected a problem with your setup of machine learning, the state index alias ["
```

Comment:
Might be best not to say "your" in the error, as the problem could be due to a bug and users might take offence that we're blaming them for it.

```suggestion
                    String msg = "Detected a problem with the internal machine learning data: the state index alias ["
```